app/assets/javascripts/govuk_publishing_components/components/step-by-step-nav.js

```diff
@@ -170,11 +181,23 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
       var contentId = thisel.querySelectorAll('.js-panel')[0].getAttribute('id')
       var titleText = title.textContent || title.innerText // IE8 fallback
 
+      if (this.$module.isGa4Enabled) {
+        var ga4JSON = {
+          event_name: 'select_content',
+          type: 'step by step',
+          text: titleText.trim(),
+          index: i + 1,
+          index_total: this.$module.totalSteps
+        }
+      }
+
+      var ga4Data = this.$module.isGa4Enabled ? "data-ga4-event='" + JSON.stringify(ga4JSON) + "'" : '' // Construct GA4 data-attributes for ga4-event-tracker.
```

Comment:
Nitpick - I'd be tempted to move this line inside the `if` statement above for clarity i.e.

```JavaScript
var ga4Data = ''
// Construct GA4 data-attributes for ga4-event-tracker
if (this.$module.isGa4Enabled) {
  var ga4JSON = {
    event_name: 'select_content',
    type: 'step by step',
    text: titleText.trim(),
    index: i + 1,
    index_total: this.$module.totalSteps
  }
  ga4Data = "data-ga4-event='" + JSON.stringify(ga4JSON) + "'" : ''
}
```